crates/turborepo-lib/src/tracing.rs

```diff
@@ -1,16 +1,107 @@
-use std::marker::PhantomData;
+use std::{marker::PhantomData, sync::Mutex};
 
 use chrono::Local;
+use is_terminal::IsTerminal;
 use owo_colors::{
     colors::{Black, Default, Red, Yellow},
     Color, OwoColorize,
 };
-use tracing::{field::Visit, Event, Level, Subscriber};
+use tracing::{field::Visit, metadata::LevelFilter, Event, Level, Subscriber};
+use tracing_appender::{
+    non_blocking::{NonBlocking, WorkerGuard},
+    rolling::RollingFileAppender,
+};
 use tracing_subscriber::{
-    fmt::{format::Writer, FmtContext, FormatEvent, FormatFields},
+    filter::Filtered,
+    fmt::{
+        self,
+        format::{DefaultFields, Writer},
+        FmtContext, FormatEvent, FormatFields,
+    },
+    prelude::*,
     registry::LookupSpan,
+    reload::{self, Error, Handle},
+    EnvFilter, Layer, Registry,
 };
 
+type StdOutLog = Filtered<
+    tracing_subscriber::fmt::Layer<Registry, DefaultFields, TurboFormatter>,
+    EnvFilter,
+    Registry,
+>;
+
+type DaemonLog = tracing_subscriber::fmt::Layer<
+    Layered,
+    DefaultFields,
+    tracing_subscriber::fmt::format::Format,
+    NonBlocking,
+>;
+
+type Layered = tracing_subscriber::layer::Layered<StdOutLog, Registry>;
+
+pub struct TurboSubscriber {
+    update: Handle<Option<DaemonLog>, Layered>,
+    guard: Mutex<Option<WorkerGuard>>,
+}
+
+impl TurboSubscriber {
+    /// Sets up the tracing subscriber, with a default stdout layer using the
+    /// TurboFormatter.
+    ///
+    /// ## Logging behaviour:
+    /// - If stdout is a terminal, we use ansi colors. Otherwise, we do not.
+    /// - If the `TURBO_LOG_VERBOSITY` env var is set, it will be used to set
+    ///   the verbosity level. Otherwise, the default is `WARN`. See the
+    ///   documentation on the RUST_LOG env var for syntax.
```

Comment:
This is critically a superset of debug, trace, warn, etc but also allows fun things like `TURBO_LOG_VERBOSITY=turborepo-lib=trace,tonic=debug,error`